src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarBidAction.tsx

```diff
@@ -104,69 +127,98 @@ export class ArtworkSidebarBidAction extends React.Component<
     const myLotStanding = artwork.myLotStanding && artwork.myLotStanding[0]
     const hasMyBids = !!(myLotStanding && myLotStanding.most_recent_bid)
 
-    if (artwork.sale.is_preview) {
+    /*
+     * Note: If the user is already registered to bid, then the sale doesn't
+     * requireIdentityVerification OR the user was manually approved by an
+     * admin. In either case, they don't need identity verification at this
+     * time.
+     */
+    const userNeedsIdentityVerification: boolean =
+      !registeredToBid &&
+      sale.requireIdentityVerification &&
+      !me?.identityVerified
+
+    if (sale.is_preview) {
       return (
-        <>
+        <div>
           {!registrationAttempted && (
-            <Button
-              width="100%"
-              size="large"
-              mt={1}
-              onClick={() => this.redirectToRegister()}
-            >
-              Register to bid
-            </Button>
+            <RegisterToBidButton onClick={this.redirectToRegister} />
           )}
+
           {registrationAttempted && !registeredToBid && (
             <Button width="100%" size="large" mt={1} disabled>
               Registration pending
             </Button>
           )}
+
```

Comment:
I realize it isn't logically possible but these buttons being listed one after another feels strange to me:
```jsx
{showButtonA && <ButtonA />}
{showButtonB && <ButtonB />}
```
It's a minor case  and fine as is, but one thing I've done in the past is something like 
```jsx
let ActionButton: React.FC
if (!registrationAttempted) {
  ActionButton = () => <RegisterToBidButton onClick={this.redirectToRegister} />
} else if (registrationAttempted && !registeredToBid) {
  ActionButton = () => (<Button width="100%" size="large" mt={1} disabled>
    Registration pending
  </Button>)
} // Ts will force the button to be defined

...
<ActionButton />
```
It's possible this is an antipattern though, I"m not sure.

Reply:
I'll take a stab at this. I do think the type narrowing you mention above is very advantageous. I'm wary of adding too much more complexity to this PR though so will timebox

Reply:
See a7f30c5 – while I didn't get any help from the Typescript compiler, I think the code is a bit easier to read regardless.